### ISSUE TITLE
Warning during compile on Elixir 1.8.0-otp-21.2.2

### ISSUE THREAD

**Issue by OvermindDL1** [ericentin/gen_state_machine#20]
It's related to the code_change spec:
```elixir
==> gen_state_machine
Compiling 3 files (.ex)
warning: invalid type annotation. When using the | operator to represent the union of types, make sure to wrap type annotations in parentheses: {:ok, state, data} | {callback_mode, state, data} | reason :: term
  lib/gen_state_machine.ex:436

Generated gen_state_machine app

```

**Comment by ericentin** [comment]
Thanks for report, but already fixed in #18 and released as v2.0.4. Please reopen if I'm mistaken. Cheers!

**Comment by OvermindDL1** [comment]
I am running this version (via `mix deps`):
```elixir
* gen_state_machine 2.0.4 (Hex package) (mix)
  locked at 2.0.4 (gen_state_machine) cc31749b
```
And I did get the above compile error in the OP.  :-)

Doesn't look like I can reopen this?

**Comment by ericentin** [comment]
You are correct, I'm not sure how it happened but a version with 2.0.4 was published without the modification to the typespec. Otherwise, it is identical to the version in the v2.0.4 tag. Must have been an accident I made around what I had checked out in git. Going to publish new v2.0.5 with the typespec change properly included. Thanks again!

**Comment by ericentin** [comment]
Ah, I think all along this was my mistake for not having released the commit in question at all! Mystery solved. :)

**Comment by ericentin** [comment]
That has now been released as v2.0.5, and I've verified it does not produce the warning from a test project with the package in its deps on Elixir v1.8.

**Comment by OvermindDL1** [comment]
> You are correct, I'm not sure how it happened but a version with 2.0.4 was published without the modification to the typespec. Otherwise, it is identical to the version in the v2.0.4 tag. Must have been an accident I made around what I had checked out in git. Going to publish new v2.0.5 with the typespec change properly included. Thanks again!

Woo, thanks much!!

> Ah, I think all along this was my mistake for not having released the commit in question at all! Mystery solved. :)

Lol, that happens to us all.  ^.^;

> That has now been released as v2.0.5, and I've verified it does not produce the warning from a test project with the package in its deps on Elixir v1.8.

Awesome, thanks much!  Updating now.  :-)
If there are any errors I'll be back inside of 5 minutes, otherwise all good.  :-)
Thanks again!!
